app/models/wish_item.rb

```diff
@@ -23,6 +26,10 @@ class WishItem < ActiveRecord::Base
 
   validate :check_finish_date, if: proc { finish_date.present? && self.finish_date_changed? }
 
+  def self.find_by_slug_or_id(param)
```

Comment:
I also defined that method for Organization, User, and Gift Item. That method returns an instance given a "slug" or an id. (The slug is used for the pretty url, for example, "luciana-reznik"